ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_acr.py

```diff
@@ -0,0 +1,299 @@
+    description:
+    - Type of object this access control record applies to.
+  chap_user:
+    required: False
+    type: str
+    description:
+    - Name for the CHAP user.
+  initiator_group:
+    required: False
+    type: str
+    description:
+    - The Initiator group name.
+  lun:
+    required: False
+    type: int
+    description:
+    - If this access control record applies to a regular volume, this attribute is the volume's LUN (Logical Unit Number).
+    If the access protocol is iSCSI, the LUN will be 0. However, if the access protocol is fibre channel, the LUN will be in the range from 0 to 2047.
+  pe_ids:
+    required: False
+    type: list
+    description:
+    - List of candidate protocol endpoints that may be used to access the Virtual volume. One of them will be selected for the access control record.
+    This field is required only when creating an access control record for a virtual volume.
+  protocol_endpoint:
+    required: False
+    type: str
+    description:
+    - Name for the protocol endpoint this access control record applies to.
+  snapshot:
+    required: False
+    type: str
+    description:
+    - Name of the snapshot this access control record applies to.
+  state:
+    required: True
+    choices:
+    - present
+    - absent
+    - create
+    type: str
+    description:
+    - Choice for access control record operation.
+  volume:
+    required: False
+    type: str
+    description:
+    - Name for the volume this access control record applies to.
+extends_documentation_fragment: hpe_nimble
+short_description: Manages a HPE Nimble Storage Access Control Record.
+version_added: 2.9
+'''
+
+EXAMPLES = r'''
+
+# If state is "create", create access control record for given volume, fails if it exist.
+# If state is "present", create access control record if not already present.
+- name: Create access control record for volume
+  hpe_nimble_acr:
+    hostname: "{{ hostname }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    volume: "{{ volume }}"
+    initiator_group: "{{ initiator_group }}"
+    state: "{{ state | default('present') }}" # fail if exist
+
+# Delete the access control record for a given volume name
+- name: Delete access control record for volume
+  hpe_nimble_acr:
+    hostname: "{{ hostname }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    volume: "{{ volume }}"
+    state: "absent" # fail if volume does not exist
+
+'''
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from nimbleclient.v1 import client
+except ImportError:
+    client = None
+import ansible_collections.hpe_nimble.array.plugins.module_utils.hpe_nimble as utils
+
+
+def create_acr(
+        client_obj,
+        initiator_group,
+        volume,
+        state,
+        **kwargs):
+
+    if utils.is_null_or_empty(initiator_group):
+        return (False, False, "Access Control Record creation failed. No initiator group provided.")
```

Comment:
I can see Access Control Record is capitalized everywhere, I think we decided not to?

Reply:
have renamed